server/hotrod/src/main/scala/org/infinispan/server/hotrod/HotRodServer.scala

```diff
@@ -247,7 +248,19 @@ class HotRodServer extends AbstractProtocolServer("HotRod") with Log {
       clientListenerRegistry.setEventMarshaller(Option(marshaller))
    }
 
-   override def stop: Unit = {
+   def addCacheStore(name: String, instance: Object): Unit = {
```

Comment:
Cache stores are normally designed for use with a single cache. Seems like that if a cache store is deployed, this is added to all the known caches. This looks risky somehow since a cache store might keep cache specific state in the implementation. This could get messy if multiple caches are using the same cache store instance.

My preference would be either: 
- a cache store to be deployed against a given cache
- or a cache store is deployed for all caches, but in this case, each cache instantiates it for its own use, so it should really get the class name and each class instantiates it for its own use.

The bigger problem here, and the reason why this feels so hacky, is that there's not a way to add cache stores to a running cache. This should be addressed properly instead of hacking around the component registry IMO.